pumpkin/src/server/translation.rs

```diff
@@ -0,0 +1,208 @@
+#![expect(dead_code)]
+
+use std::{
+    collections::HashMap,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+use pumpkin_config::TranslationConfig;
+use pumpkin_core::text::{style::Style, TextComponent, TextContent};
+use serde_json::Value;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TranslationError {
+    #[error("File cannot be opened.")]
+    InvalidFile,
+    #[error("Failed to read file. Error: {0}")]
+    FileRead(std::io::Error),
+    #[error("Invalid JSON encountered. Use only objects in the translation file.")]
+    JsonParse,
+}
+
+pub enum Translations<'a> {
+    Untranslated(TextComponent<'a>),
+    Translated(HashMap<String, TextComponent<'a>>),
+}
+
+pub fn translate<'a>(
+    config: &TranslationConfig,
+    message: &'a str,
+) -> Result<Translations<'a>, TranslationError> {
+    if !config.enabled {
+        return Ok(Translations::Untranslated(TextComponent::text(message)));
+    }
+
+    if config.client_translations {
+        return Ok(Translations::Untranslated(TextComponent {
+            content: TextContent::Translate {
+                translate: std::borrow::Cow::Borrowed(message),
+                with: vec![],
+            },
+
+            style: Style::default(),
+        }));
+    }
+
+    let path = "";
```

Comment:
Why there is an empty path ?

Reply:
empty path because i haven't set up file handling yet. Should've left a comment on that, sorry.